spectral_cube/base_class.py

```diff
@@ -688,38 +741,79 @@ def _check_beam_areas(self, threshold, mean_beam, mask=None):
 
         errormessage = ""
 
+        # Add a note when we have strict mode enabled.
+        if self.strict_beam_match:
+            strictmessage = "Strict beam match mode is enabled (a threshold of 0.0 is always enforced)." \
+                " To disable, set `cube.strict_beam_match = False`.\n"
+        else:
+            strictmessage = ""
+
         for (qtyname, qty) in (qtys.items()):
             minv = qty[mask].min()
             maxv = qty[mask].max()
-            mn = getattr(mean_beam, qtyname)
+            mn = getattr(common_beam, qtyname)
             maxdiff = (np.max(np.abs(u.Quantity((maxv-mn, minv-mn))))/mn).decompose()
 
             if isinstance(threshold, dict):
                 th = threshold[qtyname]
             else:
                 th = threshold
 
+            # If strict mode enabled, beams must match exactly.
+            if self.strict_beam_match:
+                if hasattr(th, 'unit'):
+                    th = 0. * th.unit
+                else:
+                    th = 0.
+
             if maxdiff > th:
                 errormessage += ("Beam {2}s differ by up to {0}x, which is greater"
                                  " than the threshold {1}\n".format(maxdiff,
                                                                     threshold,
                                                                     qtyname
                                                                    ))
         if errormessage != "":
-            raise ValueError(errormessage)
 
-    def mask_out_bad_beams(self, threshold, reference_beam=None,
+            if raise_error or self.strict_beam_match:
+                raise ValueError(f"{strictmessage}{errormessage}\nConvolve to a common beam before applying any"
+                                 " spectral operation.")
+
+            else:
+                warnings.warn(errormessage)
+                warnings.warn(strictmessage)
+                warnings.warn("Convolution to a common beam will be triggered at an intermediate level."
+                              " To avoid this step, first convolve the spectral-cube to a common beam size.")
```

Comment:
spectral-cube -> spectral cube or SpectralCube (with the hyphen is the package name?)